src/commands/infra/create.ts

```diff
@@ -0,0 +1,136 @@
+ *
+ * @param command Commander command object to decorate
+ */
+
+// Global vars that attain if spk infra init has been executed and the path to a bedrock template directory
+let InitIsDone: boolean = true;
+let bedrockDir: string = `${process.cwd()}/.bedrock/bedrock/cluster/environments`;
+export const createCommandDecorator = (command: commander.Command): void => {
+  command
+    .command("create")
+    .alias("c")
+    .description(
+      "Create a bedrock template based on user args and deploy infrastructure template to a provided subscription."
+    )
+    .option(
+      "-e, --environment <environment-name>",
+      "Deploy an Infra Environment from Bedrock"
+    )
+    .option(
+      "--resource-group <rg_name>",
+      "Name of resource group to deploy Bedrock Environment to"
+    )
+    .option(
+      "--cluster-name <cluster-name>",
+      "Name of the AKS cluster to deploy in environment",
+      "spk-AKScluster"
+    )
+    .option(
+      "--gitops-url <url_gitops>",
+      "URL to HLD gitops manifests to apply to AKS cluster",
+      "git@github.com:timfpark/fabrikate-cloud-native-manifests.git"
+    )
+    .option(
+      "--serviceprincipalid <sp-id>",
+      "Service Principal ID for Azure Subscription"
+    )
+    .option(
+      "--serviceprincipalsecret <sp-secret> ",
+      "Service Principal Secret for Azure Subscription"
+    )
+    .action(async opts => {
+      console.log(opts.environment);
+      try {
+        if (
+          opts.environment &&
+          opts.serviceprincipalid &&
+          opts.serviceprincipalsecret
+        ) {
+        } else {
+          logger.warn(
+            "You need to specify each of the config settings in order to run any command. Please verify you have passed an Environment, Service Principal ID, and Service Principal Secret"
+          );
+        }
+        // If Bedrock is not found, install a working path for Bedrock
+        if (!fs.existsSync(bedrockDir)) {
+          const cloneBedrock = await exec("git", [
+            "clone",
+            "https://github.com/microsoft/bedrock.git",
+            ".bedrock"
+          ]);
+        }
+        await validateInit(bedrockDir);
+        await templateInit(bedrockDir, opts.environment);
+        //await templateConfig();
+        //await templateDeploy();
+      } catch (err) {
+        logger.error("Error occurred while initializing");
+        logger.error(err);
+      }
+    });
+};
+
+export const validateInit = async (bedrockPath: string): Promise<boolean> => {
+  try {
+    // TODO: Use this function to check the state of spk infra init and attain bedrock source location
+    if (fs.existsSync(bedrockPath)) {
+      if (InitIsDone) {
+        logger.info(
+          "`spk infra init` has been successfully executed, you may now proceed to deploy Bedrock environments."
+        );
+        return true;
+        logger.info(bedrockPath);
+        process.chdir(bedrockPath);
```

Comment:
Would it be possible to change the working dir back to the original once the command finishes? Having a different working dir might break some commands down the line if we start composing them together

Reply:
reverting back to original dir